pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +119,270 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
+	}
+
+	// Check the user approval policy.
+	userApprovalPolicy, err := r.ReadUserApprovalPolicyConfig(request.Namespace)
+	if err != nil {
+		reqLogger.Error(err, "Error reading user approval policy")
+		statusError := r.setStatusFailedToReadUserApprovalPolicy(instance, err.Error())
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+
+		return reconcile.Result{}, err
+	}
+
+	// If the signup has been explicitly approved (by an admin), or the user approval policy is set to automatic,
+	// then proceed with the signup
+	if instance.Spec.Approved || userApprovalPolicy == config.UserApprovalPolicyAutomatic {
+		if instance.Spec.Approved {
+			statusError := r.setStatusApprovedByAdmin(instance, "")
+			if statusError != nil {
+				reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+				return reconcile.Result{}, statusError
+			}
+		} else {
+			statusError := r.setStatusApprovedAutomatically(instance, "")
+			if statusError != nil {
+				reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+				return reconcile.Result{}, statusError
+			}
+		}
+
+		var targetCluster string
+
+		// If a target cluster hasn't been selected, select one from the members
+		if instance.Spec.TargetCluster != "" {
+			targetCluster = instance.Spec.TargetCluster
+		} else {
+			// Automatic cluster selection
+			members := cluster.GetMemberClusters()
+			if len(members) > 0 {
+				targetCluster = members[0].Name
+			} else {
+				reqLogger.Error(err, "No member clusters found")
+				statusError := r.setStatusNoClustersAvailable(instance, "No member clusters found")
+				if statusError != nil {
+					reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+					return reconcile.Result{}, statusError
+				}
+
+				err = NewSignupError("No target clusters available")
+				return reconcile.Result{}, err
+			}
+		}
+
+		// Provision the MasterUserRecord
+		err = r.provisionMasterUserRecord(instance, targetCluster, reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		statusError := r.setStatusPendingApproval(instance, "")
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+	}
+
 	return reconcile.Result{}, nil
 }
+
+// provisionMasterUserRecord does the work of provisioning the MasterUserRecord
+func (r *ReconcileUserSignup) provisionMasterUserRecord(userSignup *toolchainv1alpha1.UserSignup, targetCluster string, logger logr.Logger) error {
+	userAccounts := []toolchainv1alpha1.UserAccountEmbedded{
+		{
+			TargetCluster: targetCluster,
+			Spec: toolchainv1alpha1.UserAccountSpec{
+				NSTemplateSet: toolchainv1alpha1.NSTemplateSetSpec{
+					Namespaces: []toolchainv1alpha1.Namespace{},
+				},
+			},
```

Comment:
This is not enough.
According to our UserAccountSpec API there are a few required fields - https://github.com/codeready-toolchain/api/blob/master///pkg/apis/toolchain/v1alpha1/useraccount_types.go#L11-L29
- UserID
- NSLimit
- NSTemplateSet

But you set NSTemplateSet only. We can set some hardcoded/default nsLimit value for now but UserID is redundant here... (still required by API).
I see a few options how to solve it:
- Keep both MUR.UserID and MUR.UserAccount[i].Spec.UserID as quick solution (and take care of it in a separate PR/issue)
- Drop UserID from MUR Spec and propagate it to every UserAccount Spec in that MUR
- Decouple UserAccount.Spec API from MUR.UserAccount[i].Spec API.

Reply:
Set values for `UserID` and `NSLimit` also, as discussed will deal with the duplicate setting of `UserID` in a separate PR.